pkg/injector/webhook.go

```diff
@@ -315,6 +316,36 @@ func (wh *mutatingWebhook) mustInject(pod *corev1.Pod, namespace string) (bool,
 	return false, nil
 }
 
+// GetPodOutboundPortExclusionList gets a list of ports to exclude from outbound sidecar interception.
+//
+// Outbound Ports are excluded from sidecar interception when:
+// 1. The pod is going to be injected with a sidecar
+// 2. The pod is explicitly annotated with a single or comma separate list of ports
+//
+// The function returns an error when it is unable to determine whether ports need to be excluded from outbound sidecar interception.
+func (wh *mutatingWebhook) GetPodOutboundPortExclusionList(pod *corev1.Pod, namespace string) ([]string, error) {
+	var ports []string
+	if inject, err := wh.mustInject(pod, namespace); err != nil || !inject {
+		log.Warn().Msgf("Outbound port exclusion request is for pod with UID %s; exclusion in Namespace %s is not permitted", pod.ObjectMeta.UID, namespace)
+		return ports, nil
+	}
```

Comment:
This check is unnecessary because `mustInject()` is invoked for the pod prior to `createPatch()`

Reply:
good find, I have removed this logic